agents/subtask_predictor.py

```diff
@@ -0,0 +1,311 @@
+import sys
+sys.path.append("/home/biswas/overcooked_ai/src/")
+from arguments import get_arguments
+from overcooked_dataset import Subtasks
+from overcooked_ai_py.mdp.overcooked_env import OvercookedEnv
+from overcooked_ai_py.mdp.overcooked_mdp import OvercookedState, OvercookedGridworld, Direction, Action
+from overcooked_ai_py.visualization.state_visualizer import StateVisualizer
+from overcooked_dataset import OvercookedDataset, Subtasks
+from state_encodings import ENCODING_SCHEMES
+from sklearn.metrics import accuracy_score
+
+from copy import deepcopy
+import numpy as np
+from pathlib import Path
+import pygame
+from pygame.locals import HWSURFACE, DOUBLEBUF, RESIZABLE, QUIT, VIDEORESIZE
+from tqdm import tqdm
+import time
+import torch as th
+import torch.nn as nn
+import torch.nn.functional as F
+from torch.utils.data import DataLoader
+from torch.distributions.categorical import Categorical
+import wandb
+import tensorflow as tf
+import torch
+import random
+def get_output_shape(model, image_dim):
+    return model(th.rand(*(image_dim))).data.shape[1:]
+
+def weights_init_(m):
+    if hasattr(m, 'weight') and m.weight is not None and len(m.weight.shape) > 2:
+        th.nn.init.xavier_uniform_(m.weight, gain=1)
+    if hasattr(m, 'bias') and m.bias is not None and isinstance(m.bias, th.Tensor):
+        th.nn.init.constant_(m.bias, 0)
+
+def merge_two_dicts(x, y):
+    z = x.copy()   # start with keys and values of x
```

Comment:
The only time you call this function, you update x anyways, so the copy is unnecessary. You can just use update directly and remove this fn